core/domain/stats_domain_test.py

```diff
@@ -197,51 +62,65 @@ def test_record_answer(self):
 
         # answer is a string
         event_services.AnswerSubmissionEventHandler.record(
-            'eid', exp_version, FIRST_STATE_NAME, self.DEFAULT_RULESPEC_STR,
-            'sid1', TIME_SPENT, PARAMS, 'answer1')
+            'eid', exp_version, FIRST_STATE_NAME, 0, 0,
+            exp_domain.EXPLICIT_CLASSIFICATION, 'sid1', TIME_SPENT, PARAMS,
+            'answer1')
         event_services.AnswerSubmissionEventHandler.record(
-            'eid', exp_version, FIRST_STATE_NAME, self.DEFAULT_RULESPEC_STR,
-            'sid2', TIME_SPENT, PARAMS, 'answer1')
+            'eid', exp_version, FIRST_STATE_NAME, 0, 1,
+            exp_domain.EXPLICIT_CLASSIFICATION, 'sid2', TIME_SPENT, PARAMS,
+            'answer1')
         # answer is a dict
         event_services.AnswerSubmissionEventHandler.record(
-            'eid', exp_version, FIRST_STATE_NAME, self.DEFAULT_RULESPEC_STR,
-            'sid1', TIME_SPENT, PARAMS, {'x': 1.0, 'y': 5.0})
+            'eid', exp_version, FIRST_STATE_NAME, 1, 0,
+            exp_domain.EXPLICIT_CLASSIFICATION, 'sid1', TIME_SPENT, PARAMS,
+            {'x': 1.0, 'y': 5.0})
         # answer is a list
         event_services.AnswerSubmissionEventHandler.record(
-            'eid', exp_version, SECOND_STATE_NAME, self.DEFAULT_RULESPEC_STR,
-            'sid3', TIME_SPENT, PARAMS, [2, 4, 8])
+            'eid', exp_version, SECOND_STATE_NAME, 2, 0,
+            exp_domain.EXPLICIT_CLASSIFICATION, 'sid3', TIME_SPENT, PARAMS,
+            [2, 4, 8])
         # answer is a unicode string
```

Comment:
It does seem relevant, I think? See lines 83-114 or so of the new version of the file -- the checks for AnswerSubmissionEventHandler.record() have been preserved.